purchase_work_acceptance/models/purchase.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.addons import decimal_precision as dp
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    wa_count = fields.Integer(
+        compute='_compute_wa',
+        string='WA count',
+        default=0,
+        store=True
+    )
+    wa_ids = fields.Many2many(
+        comodel_name='work.acceptance',
+        compute='_compute_wa',
+        string='Work Acceptances',
+        copy=False,
+        store=True,
+    )
+    wa_lines = fields.One2many(
+        comodel_name='work.acceptance.line',
+        inverse_name='purchase_line_id',
+        string="WA Lines",
+        readonly=True,
+        copy=False
+    )
+
+    @api.depends('order_line.wa_lines.wa_id')
+    def _compute_wa(self):
+        for order in self:
+            wa = self.env['work.acceptance']
```

Comment:
You don't need to declare wa object everytime. You can just use something like,
```
order.wa_ids |= line.wa_lines.mapped('wa_id')
```

BUT, to find wa_count, why don't we just,
```
order.wa_count = self.env['purchase.work.acceptance'].search_count([('purchase_id', '=', order.id)])
```